### ISSUE TITLE
Is it better to do one continuous run without --resume?

### ISSUE THREAD

**Issue by Tophness** [kohya-ss/musubi-tuner#835]
I've noticed that if I do long runs without any saving and resuming, there's generally a smooth drop in loss, and I'm not sure if it's a coincidence or not, but all my best quality models trained with sd-scripts and musubi-tuner seem to have been without resuming.

When I resume, loss always seems to start way higher than it was at the end of the last epoch it was saved at, then dip, then shoot up slightly less high, then fluctuate a bit somewhere around where it was before, but it never seems to start dropping smoothly again before I get to the end of the epoch.
I generally then save that one, stop training to run inference on it, then resume again, and the loss fluctuation starts all over again.
Is this some kind of batch variance or data shuffling that means the loss isn't reported properly in reference to previous epochs properly or something, or is it more like it's trying to recover and never getting a chance to in 1 epoch, so I shouldn't be stopping and starting all the time like that?

in case it makes a difference:
batch size 1,
--network_args "loraplus_lr_ratio=4" --lr_scheduler polynomial --lr_scheduler_power 4 --gradient_checkpointing --gradient_accumulation_steps=1 --preserve_distribution_shape --optimizer_type adamw8bit

**Comment by kohya-ss** [comment]
Since the average loss is the average of the losses over the steps equivalent to one epoch up to that point, the loss may be large after resuming until one epoch has passed. So in some cases, it may start low and go up.

Please note that the average loss after resuming cannot be compared to before resuming until one epoch has passed.

**Comment by Tophness** [comment]
> Since the average loss is the average of the losses over the steps equivalent to one epoch up to that point, the loss may be large after resuming until one epoch has passed. So in some cases, it may start low and go up.
> 
> Please note that the average loss after resuming cannot be compared to before resuming until one epoch has passed.

Thanks, so it's purely a loss reporting issue and the model itself will be fine even with constant resumes?

I just started a new training run and it was 0.068, 0.067, 0.066, consistently dropping 0.0001 each epoch.
Stopped there at 3 epochs and was like 0.0877, 0.065, 0.064, 0.063, 0.062, 0.061, 0.060.
Stopped there at 7 epochs and it started at 0.0738, then it went to 0.0623
Stopped there at 1 epochs and it started at 0.0426, then it went to 0.0621, 0.0627.
After that, I trained for another 5 epochs, and loss basically stayed around 0.0625. Sometimes rising and coming back down, but it never seems to continue dropping.
Quality of the model stopped changing once loss stopped going down after that 2nd round with 7 epochs too, and it was consistently improving right up until that. This is a dataset of 670 diverse 5 second videos.
It really seems like if I pause too soon, it basically never gets a chance to start going down, even after waiting an epoch or two.
Maybe that's a coincidence, just want to make sure I'm not wasting money if there's a possibility resume can mess things up like that.
